crates/ruma-events/src/room/message.rs

```diff
@@ -915,46 +917,103 @@ pub(crate) fn parse_markdown(text: &str) -> Option<String> {
         return None;
     }
 
+    // If the markdown contains only inline tags, i.e. if there is only one paragraph, remove the
+    // wrapping paragraph, as instructed by the spec.
+    let first_event_is_paragraph_start =
+        parser_events.front().is_some_and(|event| matches!(event, Event::Start(Tag::Paragraph)));
+    let last_event_is_paragraph_end =
+        parser_events.back().is_some_and(|event| matches!(event, Event::End(TagEnd::Paragraph)));
+
+    if first_event_is_paragraph_start && last_event_is_paragraph_end {
+        found_first_paragraph = false;
+        let has_several_blocks = parser_events.iter().any(|ref event| {
```

Comment:
At this point we have established that the first event is a paragraph-start event. I think it would thus be easier to not track the first paragraph within the `any` closure, and instead apply `.skip(1)` to the iterator before calling `any`.

Also, I'm surprised the `ref` pattern isn't generating a warning. Would you be against disabling `ref` patterns entirely in Ruma? I'm not sure why I didn't do it yet, I think they're pretty obscure and not useful enough to ever be used.

Reply:
Oh right, nice catch!

I didn't even pay attention to the `ref`, it started as a copy-paste of the iteration code above. But yeah I am for removing uses of `ref`.